subsys/net/lib/lwm2m_client_utils/CMakeLists.txt

```diff
@@ -13,3 +13,5 @@ zephyr_library_sources_ifdef(CONFIG_LWM2M_CLIENT_UTILS_FIRMWARE_UPDATE_OBJ_SUPPO
 zephyr_library_sources_ifdef(CONFIG_LWM2M_CLIENT_UTILS_CONN_MON_OBJ_SUPPORT lwm2m/lwm2m_connmon.c)
 zephyr_library_sources_ifdef(CONFIG_LWM2M_CLIENT_UTILS_DEVICE_OBJ_SUPPORT lwm2m/lwm2m_device.c)
 zephyr_library_sources_ifdef(CONFIG_LWM2M_CLIENT_UTILS_SECURITY_OBJ_SUPPORT lwm2m/lwm2m_security.c)
+zephyr_library_sources_ifdef(CONFIG_LWM2M_CLIENT_UTILS_SIGNAL_MEAS_INFO_OBJ_SUPPORT lwm2m/ecid_signal_meas_info.c)
```

Comment:
No need to check for the macro twice - for [example](https://github.com/zephyrproject-rtos/zephyr/blob/25e6803705b25602eb0b730a03a254c354bfae9e/subsys/net/lib/lwm2m/CMakeLists.txt#L33)